testdata/btf_map_init.c

```diff
@@ -0,0 +1,57 @@
+/* This file excercises the ELF loader. It is not a valid BPF program. */
+
+#include "common.h"
+
+#if __clang_major__ >= 9
+
+int __section("socket/tail") tail_1() {
+	return 42;
+}
+
+// Tail call map (program array) initialized with program pointers.
+struct {
+	__uint(type, BPF_MAP_TYPE_PROG_ARRAY);
+	__uint(key_size, sizeof(uint32_t));
+	__uint(max_entries, 2);
+	__array(values, int ());
```

Comment:
Echoing Andrii's comment from the earlier PR: does this work for `int (void*)` as well?

Reply:
Yes, ofc it works when the function signatures match. `T ()` seems to be agnostic as to what the function arguments are. The function signature isn't reflected in the resulting ELF, so from the loader side there's no difference.